src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -371,55 +364,79 @@ def check_kube_connection(configuration):
-        raise ValidationError("Helm is not installed or the helm binary is not accessible to the connectedk8s cli. Could be a permission issue.",
-                              recommendation="Ensure that you have the latest version of Helm installed on your machine and run using admin privilege. " +
-                              "Learn more at https://aka.ms/arc/k8s/onboarding-helm-install")
-    except Exception as e2:
-        telemetry.set_exception(exception=e2, fault_type=consts.Check_HelmInstallation_Fault_Type,
-                                summary='Error while verifying helm installation')
-        raise ValidationError("Error occured while verifying helm installation: " + str(e2))
-
-
-def check_helm_version(kube_config, kube_context):
-    cmd_helm_version = ["helm", "version", "--short", "--client"]
-    if kube_config:
-        cmd_helm_version.extend(["--kubeconfig", kube_config])
-    if kube_context:
-        cmd_helm_version.extend(["--kube-context", kube_context])
-    response_helm_version = Popen(cmd_helm_version, stdout=PIPE, stderr=PIPE)
-    output_helm_version, error_helm_version = response_helm_version.communicate()
-    if response_helm_version.returncode != 0:
-        telemetry.set_exception(exception=error_helm_version.decode('ascii'), fault_type=consts.Check_HelmVersion_Fault_Type,
-                                summary='Unable to determine helm version')
-        raise CLIInternalError("Unable to determine helm version: " + error_helm_version.decode("ascii"))
-    if "v2" in output_helm_version.decode("ascii"):
-        telemetry.set_exception(exception='Helm 3 not found', fault_type=consts.Helm_Version_Fault_Type,
-                                summary='Helm3 not found on the machine')
-        raise ValidationError("Helm version 3+ is required.",
-                              recommendation="Ensure that you have installed the latest version of Helm. " +
-                              "Learn more at https://aka.ms/arc/k8s/onboarding-helm-install")
-    return output_helm_version.decode('ascii')
+def install_helm_client():
+    # Return helm client path set by user
+    if os.getenv('HELM_CLIENT_PATH'):
+        return os.getenv('HELM_CLIENT_PATH')
+
+    # Fetch system related info
+    operating_system = platform.system().lower()
+    machine_type = platform.machine()
+
+    # Send machine telemetry
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.MachineType': machine_type})
+
+    # Set helm binary download & install locations
+    if(operating_system == 'windows'):
+        download_location_string = f'.azure\\helm\\{consts.HELM_VERSION}\\helm-{consts.HELM_VERSION}-{operating_system}-amd64.zip'
+        install_location_string = f'.azure\\helm\\{consts.HELM_VERSION}\\{operating_system}-amd64\\helm.exe'
+        requestUri = f'{consts.HELM_STORAGE_URL}/helm/helm-{consts.HELM_VERSION}-{operating_system}-amd64.zip'
```

Comment:
Wanted to check whether do we need to document this url for whitelisting in firewall?

Reply:
I have asked shashank about that. We will update the doc if any